File: workbook6/codewars/module2-python/exercise2/isANumber.py
# Given a string s, write a method (function) that will return true if its a valid single integer 
# or floating number or false if its not.

import re

def is_digit(s):
    # Regular expression pattern for a floating point number
    pattern = r'^-?(0|[1-9]\d*)(\.\d+)?$'
    
    # Check if the string matches the pattern
    if re.match(pattern, s):
        return True
    else:
        return False

File: workbook6/codewars/module1-javascript/exercise4/isANumber.js
//* Given a string s, write a method (function) that will return true if its a valid single integer 
//* or floating number or false if its not.

//* solution 1
// function isDigit(s) {
//     Regular expression pattern for a valid single integer or floating-point number
//     const regex = /^(\d+(\.\d*)?|\-\d+(\.\d*)?)$/;
    
//     return regex.test(s);
// }

//* solution 2
function isDigit(s) {
    return !isNaN(parseFloat(s)) && isFinite(s);
}
